doc/common_pitfalls.rst

```diff
@@ -225,3 +225,346 @@ Below are some tips on avoiding data leakage:
+    :class:`~sklearn.naive_bayes.GaussianNB` are randomized estimators.
+
+General recommendations
+-----------------------
+
+Getting reproducible results across multiple executions
+.......................................................
+
+In order to obtain reproducible (i.e. constant) results across multiple
+*program executions*, we need to remove all uses of `random_state=None`, which
+is the default. The recommended way is to declare a `rng` variable at the top
+of the program, and pass it down to any object that accepts a `random_state`
+parameter::
+
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.model_selection import train_test_split
+    >>> import numpy as np
+
+    >>> rng = np.random.RandomState(0)
+    >>> X, y = make_classification(random_state=rng)
+    >>> rf = RandomForestClassifier(random_state=rng)
+    >>> X_train, X_test, y_train, y_test = train_test_split(X, y,
+    ...                                                     random_state=rng)
+    >>> rf.fit(X_train, y_train).score(X_test, y_test)
+    0.84
+
+We are now guaranteed that the result of this script will always be 0.84, no
+matter how many times we run it. Changing the global `rng` variable to a
+different value should affect the results, as expected.
+
+It is also possible to declare the `rng` variable as an integer. This may
+however lead to less robust cross-validation results, as we will see in the
+next section.
+
+.. note::
+    We do not recommend setting the global `numpy` seed by calling
+    `np.random.seed(0)`. See `here
+    <https://stackoverflow.com/questions/5836335/consistently-create-same-random-numpy-array/5837352#comment6712034_5837352>`_
+    for a discussion.
+
+Robustness of cross-validation results
+......................................
+
+When we evaluate a randomized estimator performance by cross-validation, we
+want to make sure that the estimator can yield accurate predictions for new
+data, but we also want to make sure that the estimator is robust w.r.t. its
+random initialization. For example, we would like the random weights
+initialization of a :class:`~sklearn.linear_model.SGDCLassifier` to be
+consistently good across all folds: otherwise, when we train that estimator
+on new data, we might get unlucky and the random initialization may lead to
+bad performance. Similarly, we want a random forest to be robust w.r.t the
+set of randomly selected features that each tree will be using.
+
+For these reasons, it is preferable to evaluate the cross-validation
+preformance by letting the estimator use a different RNG on each fold. This
+is done by passing a `RandomState` instance (or None) to the estimator
+initialization.
+
+When we pass an integer, the estimator will use the same RNG on each fold: if
+we get a good (or bad) CV performance, it might just be because we got lucky
```

Comment:
Can we define what "good" or "bad" CV performance is?

Reply:
Not in absolute terms, scores are always compared relatively to another estimator / baseline

Reply:
In lieu of "we get a good (or bad) CV performance", can we say:
... if our model (estimator) performs well, as evaluated by CV, it might just be because...

Reply:
I'll use your suggestion, though I'll admit that as a non-native English speaker I can't really tell how it semantically differs from the original version